Source/automap.cpp

```diff
@@ -493,74 +888,168 @@ AutomapTile GetAutomapTypeView(Point map)
 	return GetAutomapType(map);
 }
 
+AutomapTile GetAutomapTileInDirection(Direction dir, Point map)
+{
+	switch (dir) {

```

Comment:
This function can be written simply as:
```c++
AutomapTile GetAutomapTileInDirection(Direction dir, Point map) {
  return GetAutomapTypeView(map + dir);
}
```

This is because `Direction` is convertible to `Displacement` and can be added to `Point` (https://github.com/diasurgical/devilutionX/blob/0f3c74af339d9b055e9e4567b7eb1d568f66539a/Source/engine/point.hpp#L64-L67)

With that in mind you may not need a function at all

Reply:
Awesome!